docs/emitters/code-generation/server/server-code-generation.md

```diff
@@ -0,0 +1,375 @@
+
+```yaml
+emit:
+  - "@typespec/openapi3"
+  - "@typespec/http-client-csharp"
+  - "@typespec/http-client-js"
+  - "@typespec/http-server-js"
+  - "@typespec/http-server-csharp"
+options:
+  "@typespec/openapi3":
+    emitter-output-dir: "{project-root}/openapi/"
+  "@typespec/http-client-csharp":
+    emitter-output-dir: "{project-root}/clients/csharp"
+  "@typespec/http-client-js":
+    emitter-output-dir: "{project-root}/clients/javascript"
+  "@typespec/http-server-csharp":
+    emitter-output-dir: "{project-root}/servers/aspnet/generated"
+  "@typespec/http-server-js":
+    emitter-output-dir: "{project-root}/servers/javascript"
+```
+
+**Explanation:**
+
+*   **`emit` section:** This section lists the *emitters* that you want to use. An emitter is a plugin that takes your TypeSpec code and generates output in a specific format (e.g., OpenAPI, C# client, JavaScript server).
+    *   `"@typespec/openapi3"`: Generates an OpenAPI 3.0 specification.
+    *   `"@typespec/http-client-csharp"`: Generates a C# client SDK.
+    *   `"@typespec/http-client-js"`: Generates a JavaScript client SDK.
+    *   `"@typespec/http-server-csharp"`: Generates a C# (ASP.NET Core) server stub.
+    *   `"@typespec/http-server-js"`: Generates a JavaScript (Node.js/Express) server stub.
+*   **`options` section:** This section configures the emitters. For each emitter, you can specify options like the output directory.
+    *   `emitter-output-dir`: This option specifies where the generated code should be placed. The `{project-root}` variable refers to the root directory of your TypeSpec project.
+
+**Important:** Make sure you have installed all the necessary emitters:
+```bash
+npm install -D @typespec/openapi3 @typespec/http-client-csharp @typespec/http-client-js @typespec/http-server-csharp @typespec/http-server-js
+```
+
+## Using the cli to generate server stubs
+
+1.  **Create a TypeSpec project:** If you don't already have one, create a new TypeSpec project using `tsp init`.
+
+    ```bash
+    mkdir my-api
+    cd my-api
+    tsp init
+    ```
+
+    During `tsp init`, you will be prompted to select different emitters, including server and client. If you have the necessary dependencies installed, TypeSpec can automatically populate your `tspconfig.yaml` file.
+
+2.  **Define your api:** Write your API specification in a `.tsp` file (e.g., `main.tsp`). This is where you define your models, operations, and routes.
+
+3.  **Compile your TypeSpec:** Use the `tsp compile` command to generate server stubs and other artifacts based on your specification.
+
+    ```bash
+    tsp compile .
+    ```
+
+    The compiler will output the generated server stubs to the directories specified in your `tspconfig.yaml` file (e.g., `servers/aspnet/generated` for C#).
+
+4.  **Project structure:** After compiling, you will typically see the following structure:
+
+    ```text
```

Comment:
See startlight filetree component to have a better presentation here https://starlight.astro.build/components/file-tree/, we use it in a few other places in the codebase too

Reply:
same for all the other dir examples

Reply:
done